Library/Homebrew/formula_installer.rb

```diff
@@ -292,6 +292,9 @@ def install
     if pour_bottle?(warn: true)
       begin
         pour
+      rescue StageFileMissingError
```

Comment:
In what situations does this occur?

Reply:
Oops, sorry. I should have been more clear about this. This is needed for when `formula#install` is invoked. We check if `fetch` has already happened for the formula, and if not, we raise this error which will then fetch.

It wasn't needed before, because fetching was happening inside of `formula_installer#pour`

This is the solution to: 

https://github.com/Homebrew/brew/pull/7124#discussion_r410060270

Reply:
Gotcha, so it's only needed for tests?

Reply:
Hm, after checking out the code more, I believe that is correct in that is only used for tests. I only see `#formula.install` being called inside of `build.rb` and that is only called from within `formula_installer`. The error is only being raised for pouring.